cloud/blockstore/libs/endpoints/endpoint_manager_ut.cpp

```diff
@@ -165,10 +165,78 @@ struct TMockSessionManager final: public ISessionManager
 
 ////////////////////////////////////////////////////////////////////////////////
 
+class TSlowDevice final: public NBD::IDevice
+{
+public:
+    explicit TSlowDevice(std::weak_ptr<::NCloud::TExecutor> executor)
+        : Executor(std::move(executor))
+    {}
+
+    TExecutorPtr GetExecutor()
+    {
+        if (Executor.expired()) {
+            return {};
+        }
+
+        return Executor.lock();
+    }
+
+    NThreading::TFuture<NProto::TError> StubFunction()
+    {
+        auto executor = GetExecutor();
+        if (!executor) {
+            return NThreading::MakeFuture(MakeError(S_OK));
+        }
+
+        return executor->Execute(
```

Comment:
not sure I understand why this is slow or helps to delay StartEndpoint/RestoreEndpoint call
why not just return external promise in the  `NThreading::TFuture<NProto::TError> Start()` and then set it to value after validating second call was delayed ?